accounts/abi/bind/bindv2.go

```diff
@@ -0,0 +1,311 @@
+	contracts map[string]*tmplContractV2
+
+	// structs is the map of all redeclared structs shared by passed contracts.
+	structs map[string]*tmplStruct
+
+	// aliases is a map for renaming instances of named events/functions/errors to specified values
+	aliases map[string]string
+}
+
+// registerIdentifier applies alias renaming, name normalization (conversion to camel case), and registers the normalized
+// name in the specified identifier map.  It returns an error if the normalized name already exists in the map.
+func (b *contractBinder) registerIdentifier(identifiers map[string]bool, original string) (normalized string, err error) {
+	normalized = methodNormalizer(alias(b.binder.aliases, original))
+	// Name shouldn't start with a digit. It will make the generated code invalid.
+	if len(normalized) > 0 && unicode.IsDigit(rune(normalized[0])) {
+		normalized = fmt.Sprintf("E%s", normalized)
+		normalized = abi.ResolveNameConflict(normalized, func(name string) bool {
+			_, ok := identifiers[name]
+			return ok
+		})
+	}
+
+	if _, ok := identifiers[normalized]; ok {
+		return "", fmt.Errorf("duplicate symbol '%s'", normalized)
+	}
+	identifiers[normalized] = true
+	return normalized, nil
+}
+
+// RegisterCallIdentifier applies registerIdentifier for contract methods.
+func (b *contractBinder) RegisterCallIdentifier(id string) (string, error) {
+	return b.registerIdentifier(b.callIdentifiers, id)
+}
+
+// RegisterEventIdentifier applies registerIdentifier for contract events.
+func (b *contractBinder) RegisterEventIdentifier(id string) (string, error) {
+	return b.registerIdentifier(b.eventIdentifiers, id)
+}
+
+// RegisterErrorIdentifier applies registerIdentifier for contract errors.
+func (b *contractBinder) RegisterErrorIdentifier(id string) (string, error) {
+	return b.registerIdentifier(b.errorIdentifiers, id)
+}
+
+// BindStructType register the type to be emitted as a struct in the
+// bindings.
+func (b *binder) BindStructType(typ abi.Type) {
+	bindStructType(typ, b.structs)
+}
+
+// contractBinder holds state for binding of a single contract
+type contractBinder struct {
+	binder *binder
+	calls  map[string]*tmplMethod
+	events map[string]*tmplEvent
+	errors map[string]*tmplError
+
+	callIdentifiers  map[string]bool
+	eventIdentifiers map[string]bool
+	errorIdentifiers map[string]bool
+}
+
+// bindMethod registers a method to be emitted in the bindings.
+// The name, inputs and outputs are normalized.  If any inputs are
+// struct-type their structs are registered to be emitted in the bindings.
+// Any methods that return more than one output have their result coalesced
+// into a struct.
+func (cb *contractBinder) bindMethod(original abi.Method) error {
+	normalized := original
+	normalizedName, err := cb.RegisterCallIdentifier(original.Name)
+	if err != nil {
+		return err
+	}
+
+	normalized.Name = normalizedName
+	normalized.Inputs = make([]abi.Argument, len(original.Inputs))
+	copy(normalized.Inputs, original.Inputs)
```

Comment:
```suggestion
	normalized.Inputs = slices.Clone(original.Inputs)
```
And same for outputs.